GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java

```diff
@@ -665,27 +671,73 @@ private void writeAnswer(
                 break;
 
             case NUMBER:
-                val = value.trim();
-                cells.add(val);
+                cells.add(value);
                 break;
 
+            // Two different formats:
+            // old: Foo|Bar|Baz (never contains code)
+            // new: [{"name": "Foo", "code": "A"}, {"name": "Bar", "code": "B"} ... ]
             case CASCADE:
+                List<Map<String, String>> parts = new ArrayList<>();
+
+                if (value.startsWith("[")) {
+                    try {
+                        parts = OBJECT_MAPPER.readValue(value,
+                                new TypeReference<List<Map<String, String>>>() {
+                                });
+                    } catch (IOException e) {
+                        log.warn("Unable to parse CASCADE response - " + value, e);
+                    }
+                } else {
+                    for (String part : value.split("\\|")) {
```

Comment:
I wanted to build a full in-memory representation of the response similar to the one we would get if the response was JSON encoded. I handle the `levelCount` issue for both response formats below.